caravel/assets/javascripts/explorev2/index.jsx

```diff
@@ -17,15 +17,21 @@ const bootstrappedState = Object.assign(initialState, {
   datasourceId: parseInt(bootstrapData.datasource_id, 10),
   datasourceType: bootstrapData.datasource_type,
   sliceName: bootstrapData.viz.form_data.slice_name,
-  sliceId: bootstrapData.viz.form_data.slice_id,
-  vizType: bootstrapData.viz.form_data.viz_type,
-  timeColumn: bootstrapData.viz.form_data.granularity_sqla,
-  timeGrain: bootstrapData.viz.form_data.time_grain_sqla,
-  metrics: [bootstrapData.viz.form_data.metrics].map((m) => ({ value: m, label: m })),
-  since: bootstrapData.viz.form_data.since,
-  until: bootstrapData.viz.form_data.until,
-  havingClause: bootstrapData.viz.form_data.having,
-  whereClause: bootstrapData.viz.form_data.where,
+  viz: {
+    formData: {
+      sliceId: bootstrapData.viz.form_data.slice_id,
```

Comment:
@bkyryliuk what do you mean about making sure the data complies to the UI expectations? can you clarify?